processor/src/main/java/org/mapstruct/ap/internal/util/workarounds/SpecificCompilerWorkarounds.java

```diff
@@ -115,4 +114,34 @@ public static TypeElement replaceTypeElementIfNecessary(Elements elementUtils, T
         }
         return element;
     }
+
+    /**
+     * Workaround for Bugs in the Eclipse implementation of {@link Types#asMemberOf(DeclaredType, Element)}.
+     *
+     * @see <a href="https://bugs.eclipse.org/bugs/show_bug.cgi?id=382590">Eclipse Bug 382590</a>
+     * @see <a href="https://bugs.eclipse.org/bugs/show_bug.cgi?id=481555">Eclipse Bug 481555</a>
+     */
+    static TypeMirror asMemberOf(Types typeUtils, ProcessingEnvironment env, DeclaredType containing, Element element) {
+        TypeMirror result = null;
+        Exception lastException = null;
+        try {
+            try {
+                result = typeUtils.asMemberOf( containing, element );
+            }
+            catch ( IllegalArgumentException e ) {
```

Comment:
Ah, sorry for missing that from an earlier comment :/

I would prefer to do it for 1.0, hiding exceptions is no good practice. But it's ok to do it separately.